openhcl/hcl/src/ioctl/snp.rs

```diff
@@ -22,14 +23,13 @@ use hvdef::HV_PAGE_SIZE;
 use memory_range::MemoryRange;
 use sidecar_client::SidecarVp;
 use std::os::fd::AsRawFd;
-use std::ptr::NonNull;
 use thiserror::Error;
 use x86defs::snp::SevRmpAdjust;
 use x86defs::snp::SevVmsa;
 
 /// Runner backing for SNP partitions.
-pub struct Snp {
-    vmsa: VtlArray<NonNull<SevVmsa>, 2>,
+pub struct Snp<'a> {
+    vmsa: &'a VtlArray<MappedPage<SevVmsa>, 2>,
```

Comment:
Let's try to keep this as `VtlArray<&'a UnsafeCell<SevVmsa, 2>>`.

I.e., let's try to get the reference inside, to avoid a double deref.

Reply:
I guess we have `each_ref` for that.